### PR TITLE
editoast: make stdcm request validation tests parameterized

### LINKED ISSUE #16814: Parameterize `stdcm_request_mass_validation` and `stdcm_request_length_validation`
### Description and goal

Follow-up on https://github.com/OpenRailAssociation/osrd/pull/16556#discussion_r3266569247.

The two following tests looks alike and could be transformed into a single parameterized tests.
- `stdcm_request_mass_validation`
- `stdcm_request_length_validation`

We could also add another test condition about `max_speed` which was introduced in #16545.

### Acceptance criteria

There is one parameterized test instead of two tests.

### PR DESCRIPTION
**Pull request by woshilapin**
Closes https://github.com/OpenRailAssociation/osrd/issues/16814

… and add a test for the validation of maximum speed (second commit)
… and fix the code for this third test (still second commit)

### CODE REVIEW

File: editoast/src/views/timetable/stdcm.rs [hunk 1]
```diff
@@ -1084,55 +1084,33 @@ mod tests {
...
-
+    #[case::incorrect_mass(
+        Some(80_000.0),
+        None,
+        None,
+        "editoast:stdcm:InvalidConsistMass",
+        "expected_min",
+        900_000.0
+    )]
     #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
```
Comment (hhirtz) [inline editoast/src/views/timetable/stdcm.rs]: duplicate `tokio::test` attribute macro?

**Review by hhirtz — APPROVED** [review]
nit: the second commit message could focus on the bug (with a bonus explanation) instead of the test

File: editoast/src/views/timetable/stdcm/request.rs [hunk 2]
```diff
@@ -398,8 +398,8 @@ impl ConsistConfiguration {
         let consist_max_speed = consist_max_speed.floor::<meter_per_second>();
 
         if let Some(request_max_speed) = self.max_speed
-            && request_max_speed < quantities::Velocity::new::<meter_per_second>(0.0)
-            && request_max_speed > consist_max_speed
+            && (request_max_speed < quantities::Velocity::new::<meter_per_second>(0.0)
+                || request_max_speed > consist_max_speed)
```
Comment (Erashin) [inline editoast/src/views/timetable/stdcm/request.rs:402]: Good catch!

File: editoast/src/views/timetable/stdcm.rs [hunk 3]
```diff
@@ -1103,6 +1103,15 @@ mod tests {
         "expected_min",
         400.0
     )]
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
```
Comment (Erashin) [inline editoast/src/views/timetable/stdcm.rs]: Ditto. We probably should only need one `tokio::test` for the whole unit test.

**Review by Erashin — APPROVED** [review]
Thx for that!

File: editoast/src/views/timetable/stdcm.rs (hunk 1 above)
Comment (woshilapin) [reply editoast/src/views/timetable/stdcm.rs]: Well, they are not really duplicate, since each `#[case]` is applied on a `async` function, which need the corresponding `#[tokio::test]`. But it made me realized it is all about the order of the decorators. Inverting them (first `tokio::test`), makes the function non-async for `rstest` to consume.

So fixed.

(see also https://github.com/OpenRailAssociation/osrd/pull/16924).

File: editoast/src/views/timetable/stdcm.rs (hunk 3 above)
Comment (woshilapin) [reply editoast/src/views/timetable/stdcm.rs]: See https://github.com/OpenRailAssociation/osrd/pull/16895#discussion_r3307185708

**Review by leovalais — APPROVED** [review]
